types/scte224v20200407/sctestructures.go

```diff
@@ -0,0 +1,325 @@
+	Match           Match     `xml:"match,attr,omitempty" json:"match,omitempty"`
+	SignalTolerance Duration  `xml:"signalTolerance,attr,omitempty" json:"signalTolerance,omitempty"`
+	Assertions      []*Assert `xml:"http://www.scte.org/schemas/224 Assert,omitempty" json:"assertions,omitempty"`
+	Schema          string    `xml:"schema,attr,omitempty" json:"schema,omitempty"`
+}
+
+// Custom unmarshalling to provide default value to "schema" attribute, when unavailable
+func (ms *MatchSignal) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	/**
+	* https://pkg.go.dev/encoding/xml#Unmarshaler
+	* One common implementation strategy is to unmarshal into a separate value with a layout matching the expected XML using d.DecodeElement,
+	* and then to copy the data from that value into the receiver.
+	* this is to prevent recursive unmarshaling ending in an infinite loop
+	 */
+	type dupMatchSignalType MatchSignal
+	matchSignal := dupMatchSignalType{}
+
+	err := d.DecodeElement(&matchSignal, &start)
+	if err != nil {
+		return err
+	}
+
+	if strings.Trim(matchSignal.Schema, " ") == "" {
+		// set the default value
+		matchSignal.Schema = matchSignalSchemaDefault
+	}
+
+	*ms = MatchSignal(matchSignal)
+	return nil
+}
+
+type Match string
+
+//	Returns true if the value of this enumerated Match is "ALL".
+func (me Match) IsAll() bool { return me == "ALL" }
+
+//	Returns true if the value of this enumerated Match is "ANY".
+func (me Match) IsAny() bool { return me == "ANY" }
+
+//	Returns true if the value of this enumerated Match is "NONE".
+func (me Match) IsNone() bool { return me == "NONE" }
+
+type Assert struct {
+	XMLName     xml.Name `xml:"http://www.scte.org/schemas/224 Assert" json:"-"`
+	Declaration string   `xml:",chardata" json:"declaration,omitempty"`
+}
+
+//********************* Audience Types *************************//
+//Table 11
+type Policy struct {
+	ReusableType
+	XMLName        xml.Name         `xml:"http://www.scte.org/schemas/224 Policy" json:"-"`
+	ViewingPolicys []*ViewingPolicy `xml:"http://www.scte.org/schemas/224 ViewingPolicy,omitempty" json:"viewingPolicys,omitempty"`
+}
+
+//Table 12
+type ViewingPolicy struct {
+	ReusableType
+	XMLName              xml.Name                    `xml:"http://www.scte.org/schemas/224 ViewingPolicy" json:"-"`
+	Audience             *Audience                   `xml:"http://www.scte.org/schemas/224 Audience,omitempty" json:"audience,omitempty"`
+	SignalPointDeletion  *SignalPointDeletionAction  `xml:"urn:scte:224:action SignalPointDeletion,omitempty" json:"signalPointDeletion,omitempty"`
+	SignalPointInsertion *SignalPointInsertionAction `xml:"urn:scte:224:action SignalPointInsertion,omitempty" json:"signalPointInsertion,omitempty"`
+	Content              *ContentAction              `xml:"urn:scte:224:action Content,omitempty" json:"content,omitempty"`
+	Allocation           *Allocation                 `xml:"urn:scte:224:action Allocation,omitempty" json:"Allocation,omitempty"`
```

Comment:
`Allocation` (and all its nested types) are a new addition

Reply:
Followed the convention of unmarshalling into a pointer type rather than a value type. I assume it's because, it's easier to pass structs around with pointer fields, because large structs are not copied over and over in memory?

Reply:
no, the pointers are to support the omitempty mechanism and avoid zero value subelements

Reply:
Okay, got it.